### PR TITLE
Fix incorrect payment amount being sent to Objects API

### LINKED ISSUE #4425: payment adds unsuccessful payments tot amount.
### Product versie / Product version

2.6.7

### Customer reference

Utrecht 243

### Omschrijf het probleem / Describe the bug

When a payment is retried multiple times, each amount get's added to the total when 1 of these tries is successful Resulting in the wrong payment.amount.

Example where only 1110 was paid, the rest never completed.
payment.amount was 4 X EUR 19,90 = EUR 79,60

image

### PR DESCRIPTION
**Pull request by sergei-maertens**
Closes #4425 

**Changes**

* `payment.amount` is now derived from `submission.price` instead of `SubmissionPayment` record aggregation. The documentation describes this field as being the cost of a submission, not the sum of all payment attempts.
* The submission variable for payment amount is updated accordingly.
* Removed (confusing and dangerous) queryset method to aggregate the total amount in submission payments.

**Checklist**

Check off the items that are completed or not relevant.

- Impact on features

  - [x] Checked copying a form
  - [x] Checked import/export of a form
  - [x] Config checks in the configuration overview admin page
  - [x] Problem detection in the admin email digest is handled

- Release management

  - [x] I have labelled the PR as "needs-backport" accordingly

- I have updated the translations assets (you do NOT need to provide translations)

  - [x] Ran `./bin/makemessages_js.sh`
  - [x] Ran `./bin/compilemessages_js.sh`

- Commit hygiene

  - [x] Commit messages refer to the relevant Github issue
  - [x] Commit messages explain the "why" of change, not the how

### CODE REVIEW

File: src/openforms/registrations/contrib/objects_api/submission_registration.py [hunk 1]
```diff
@@ -273,9 +274,11 @@ class ObjectsAPIV1Handler(ObjectsAPIRegistrationHandler[RegistrationOptionsV1]):
 
     @staticmethod
     def get_payment_context_data(submission: Submission) -> dict[str, Any]:
+        price = submission.price
```
Comment (vaszig) [inline src/openforms/registrations/contrib/objects_api/submission_registration.py:277]: Is there a possibility for the submission to be `None` here?

**Review by vaszig — APPROVED** [review]
Except for the CI and a small remark-question, it seems good to me, I don't know if you want someone else to have a look as well.

File: src/openforms/registrations/contrib/objects_api/submission_registration.py (hunk 1 above)
Comment (sergei-maertens) [reply src/openforms/registrations/contrib/objects_api/submission_registration.py:277]: no :) otherwise the type hint should have been `Submission | None`, but the previous code already used `submission.payments` so that would've been broken in the first place

File: src/openforms/logging/templates/logging/events/payment_flow_start.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
 {% load i18n %}
 {% blocktrans trimmed with plugin=log.fmt_plugin order_id=log.extra_data.payment_order_id lead=log.fmt_lead %}
    {{ lead }}: Payment plugin {{ plugin }} order '{{ order_id }}' started browser flow.
-{% endblocktrans %}
+{% endblocktrans %}{% if log.exra_data.from_email %} {% trans "Flow entered via email link." %}{% endif %}
```
Comment (stevenbal) [inline src/openforms/logging/templates/logging/events/payment_flow_start.txt:4]:
```suggestion
{% endblocktrans %}{% if log.extra_data.from_email %} {% trans "Flow entered via email link." %}{% endif %}
```

File: src/openforms/registrations/contrib/objects_api/registration_variables.py [hunk 3]
```diff
@@ -62,12 +62,14 @@ def get_initial_value(self, submission: Submission | None = None):
 @register("payment_amount")
 class PaymentAmount(BaseStaticVariable):
     name = _("Payment amount")
-    data_type = FormVariableDataTypes.string
+    data_type = FormVariableDataTypes.float
```
Comment (stevenbal) [inline src/openforms/registrations/contrib/objects_api/registration_variables.py:65]: I don't know enough about this, but why was this changed from string to float? Or was this incorrectly marked as string to begin with?

**Review by stevenbal — COMMENTED** [review]
I was too slow :open_mouth:

File: src/openforms/registrations/contrib/objects_api/registration_variables.py (hunk 3 above)
Comment (sergei-maertens) [reply src/openforms/registrations/contrib/objects_api/registration_variables.py:65]: was incorrectly marked as string, since clearly floats (and `None`) were being returned :)

File: src/openforms/logging/templates/logging/events/payment_flow_start.txt (hunk 2 above)
Comment (sergei-maertens) [reply src/openforms/logging/templates/logging/events/payment_flow_start.txt:4]: shit - will fixup!
